### ISSUE TITLE
Allow OutFile when the module is none

### ISSUE THREAD

**Issue by daniel-zazula** [Microsoft/TypeScript#26791]
## Due diligence
Before filling this suggestion I:

- Searched for something similar, on the Issues Page, using multiple combinations of the keywords bellow, including closed issues.
- Searched for this suggestion in the FAQ, including using the browser's word search. 

## Search Terms

outfile, module, none

## Suggestion

Allow the OutFile option when the Module option is set to "none".

It makes sense to block the OutFile option for projects using certain module systems, because their individual modules/files can't be merged into a single module/file. But why block OutFile when not using any module system?
Wouldn't it require just the concatenation of the .js files generated by the compiler? Or am I missing something?

## Use Cases

This would be specially useful when converting old (and/or badly coded) web projects to TypeScript, where you have big .js files and you want to break into a bunch smaller .ts files, but you still need to generate a single .js file for the page to load.
I know I can do this with tools like Gulp and Webpack, but that requires a lot more changes to old web projects.
This would also allow using multiple tsconfig files (inheriting from a master config) to generate bundles.

## Examples

On the tsconfig.json:

1. Configure the project to include multiple ts files (using the include and/or exclude options).
2. Set the compilerOptions -> outFile option.

Expected result:
During compilation, the file specified in outFile is created/updated with the javascript code generated from all the .ts files included in the project.
In other words, the same thing that would happen if you specify a module system that allows the outFile, but don't use any module import or export on your .ts files.

## Checklist

My suggestion meets these guidelines:
* [x] This wouldn't be a breaking change in existing TypeScript / JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. new expression-level syntax)

**Comment by RyanCavanaugh** [comment]
What do you mean? This is already legal.

**Comment by GongT** [comment]
😋You can try my new project: https://github.com/GongT/export-all-in-one

**Comment by daniel-zazula** [comment]
Indeed @RyanCavanaugh, I just created a project using TypeScript 3.0 and it did work. I forgot to try that before opening the issue. So much for due diligence.
But I do know it wasn't working before, on a previous project of mine. I don't know if that was because of the Typescript version (it was 2.6 I think) or because of something in my code (maybe there was an import statement somewhere).
In any case, consider this issue closed.
